src/routes/products/products.js

```diff
@@ -0,0 +1,171 @@
+            getProduct(request, response);
+            break;
+        case "/products/?category":
+            getCategory(request, response);
+            break;
+        default:
+            console.log("Nothing");
+    }
+
+
+
+};
+
+const getProducts = (request, response) => {
+    if (request.query.ids !== undefined || request.params.id !== undefined){
+      response.writeHead (200, {"Contend-type" : "application/json"});
+       response.write(JSON.stringify(getProductById(request.query.ids || request.params.id)));
+       response.end();
+   } else if (request.query.category !== undefined){
+        response.writeHead (200, {"Contend-type" : "application/json"});
+        response.write(JSON.stringify(getElementByCategory(request.query.category)));
+        response.end();
+    }
+   else {
+       response.writeHead(200, {"Content-type" : "application/json"});
+       response.write(JSON.stringify(getAllProducts()));
+       response.end();
+   }
+
+};
+
+function getProductById(id) {
+    const productsFile = path.join(__dirname, '../../../', '/src/db/products', '/all-products.json');
+    const products = fs.readFileSync(productsFile);
+
+    const responseObj = {
+        "status": "success",
+        "products": getElementId(id)
+    };
+    return responseObj;
+
+}
+
+function getAllProducts() {
+    const productsFile = path.join(__dirname, '../../../',  '/src/db/products', '/all-products.json');
+    const products = fs.readFileSync(productsFile);
+    const allProductsJS = JSON.parse(products);
+    return allProductsJS;
+}
+
+const getIdProduct = url => {
+    const lastIndex = url.lastIndexOf('=');
+
+    if(lastIndex !== -1){
+        return url.slice (lastIndex +1);
+    }
+
+};
+
+const getProduct = (request, response) => {
+    const productsFile = path.join(__dirname, '../../../', '/src/db/products', '/all-products.json');
+    const products = fs.readFileSync(productsFile);
+
+    const responseObj = {
+        "status": "success",
+        "products": getElementId(request.params.id)
+    };
+
+  response.writeHead (200, {"Contend-type" : "application/json"});
+    response.write(JSON.stringify(responseObj));
+  response.end();
+
+
+};
+
+function getElementId(id) {
+    const productsFile = path.join(__dirname, '../../../',  '/src/db/products', '/all-products.json');
+    const products = fs.readFileSync(productsFile);
+    const allProductsJS = JSON.parse(products);
+    const objectItems = [];
+    const idArray = id.split(',');
+
+    for (let idNumber of idArray){
+        for (let productItem of allProductsJS){
+            idNumber = Number(idNumber);
+            if(idNumber === productItem.id){
+                objectItems.push(productItem);
+            }
+        }
+    }
+    return objectItems;
+
+}
+
+function getElementByCategory(category) {
+    const productsFile = path.join(__dirname, '../../../',  '/src/db/products', '/all-products.json');
+    const products = fs.readFileSync(productsFile);
```

Comment:
do not use readFileSync in project many times

its duplicates 

remove duplicates